node/core/pvf/src/worker_common.rs

```diff
@@ -178,16 +177,19 @@ pub async fn tmpfile(prefix: &str) -> io::Result<PathBuf> {
 
 pub fn worker_event_loop<F, Fut>(debug_id: &'static str, socket_path: &str, mut event_loop: F)
 where
-	F: FnMut(UnixStream) -> Fut,
+	F: FnMut(Handle, UnixStream) -> Fut,
 	Fut: futures::Future<Output = io::Result<Never>>,
 {
-	let err = async_std::task::block_on::<_, io::Result<Never>>(async move {
-		let stream = UnixStream::connect(socket_path).await?;
-		let _ = async_std::fs::remove_file(socket_path).await;
-
-		event_loop(stream).await
-	})
-	.unwrap_err(); // it's never `Ok` because it's `Ok(Never)`
+	let rt = Runtime::new().expect("Creates tokio runtime");
```

Comment:
1. This should either explain why panicking here is OK or send an error to the host, `expect` still doesn't follow the qed rule.
2. `Runtime::new` defaults to `Builder::new(Kind::MultiThread, 61, 61)`. I believe we would want to configure it.

Reply:
1. I based that on usage in `substrate` which just says "Creates tokio runtime". I don't know if a proof or sending back an error is possible here, but I added an explanation.

2. I assumed those numbers referred to threads, which would make sense to configure. But looks like those numbers are `global_queue_interval` and `event_interval`. What do we want to configure here?

```rs
pub(crate) fn new(kind: Kind, global_queue_interval: u32, event_interval: u32) -> Builder {
```

For threads, I scoured the [docs](https://docs.rs/tokio/latest/tokio/runtime/index.html) but couldn't find any info on the default number of threads spawned.

Reply:
I found it:

```rs
// Default to lazy auto-detection (one thread per CPU core)
worker_threads: None,
```

Not sure if this includes blocking threads though. There's a `max_blocking_threads` option which I don't see a reason to change.